dozer-spring-boot-auto-configuration/pom.xml

```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright 2005-2017 Dozer Project
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>com.github.dozermapper</groupId>
+        <artifactId>dozer-plugins-parent</artifactId>
+        <version>6.2.0-SNAPSHOT</version>
+        <relativePath>../plugins-parent</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>dozer-spring-boot-auto-configuration</artifactId>
+    <packaging>pom</packaging>
+    <name>Spring Boot AutoConfiguration :: Dozer</name>
+
+    <dependencyManagement>
+        <dependencies> <!-- The parent should provide all that -->
+            <dependency>
+                <groupId>org.springframework.boot</groupId>
+                <artifactId>spring-boot-dependencies</artifactId>
+                <version>1.5.8.RELEASE</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.github.dozermapper</groupId>
+            <artifactId>dozer-spring</artifactId>
+            <version>${project.version}</version>
+            <optional>true</optional>
```

Comment:
whys this optional?

Reply:
Because spring boot documentation [recommends](https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-developing-auto-configuration.html#boot-features-custom-starter-module-autoconfigure) to do like this just to remove transitive dependencies. It is not an auto-configuration module's responsibility to provide needed libraries. It is up to the project dependencies. If there is no proper dependency autoconfiguration will be just ignored.

Reply:
Agree that providing dependencies is not a responsibility of autoconfiguration. But it is of the starter, the same documentation states:
> The starter module that provides a dependency to the autoconfigure module as well as the library and any additional dependencies that are typically useful. In a nutshell, adding the starter should be enough to start using that library.
...
Its only purpose is to provide the necessary dependencies to work with the library; see it as an opinionated view of what is required to get started.

Thus I would recommend to include autoconfiguration and dozer-spring as compile-time dependencies to the starter module.  

And then I would also switch dependencies in autoconfiguration module from `optional` to `provided`, because `optional` has different semantic in Maven, basically meaning "dependencies required by optional feature", while autoconfiguration will not work at all unless Dozer library is _provided_.